toolslm/download.py

```diff
@@ -90,8 +91,11 @@ def find_docs(url):
 # %% ../03_download.ipynb 19
 def read_docs(url, optional=False, n_workers=None, rm_comments=True, rm_details=True):
     "If available, return LLM-friendly llms.txt context or markdown file response for `url`"
-    url = find_docs(url)
-    if not url: return
-    if url.endswith('/llms.txt'): res = get_llmstxt(url, optional=optional, n_workers=n_workers)
-    else: res = get(url).text
+    fn = find_docs(url)
+    if not fn:
+        parsed_url = urlparse(url)
+        if parsed_url.path == '/' or not parsed_url.path: return None
+        return read_docs(urljoin(url, '..'), optional, n_workers, rm_comments, rm_details)
```

Comment:
Does this halt in the case that there's no match? See this discussion: https://gist.github.com/jph00/2dea2d43da3dd824377b1990c34c2aa7

Reply:
> Does this halt in the case that there's no match? See this discussion: https://gist.github.com/jph00/2dea2d43da3dd824377b1990c34c2aa7

Yes, it does halt.  `urljoin(url, "..")` is analogous to `cd ..`

these tests pass - should I add them?

```
test_eq(urljoin("https://llmstxt.org/tmp", ".."), "https://llmstxt.org/")
test_eq(urljoin("https://llmstxt.org/", ".."), "https://llmstxt.org/")
```